File: models/Gpt-3.5-turbo/cpp/code/problems/geometry/Polygon_Area.cpp

#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> x(n);
    vector<int> y(n);

    for (int i = 0; i < n; i++) {
        cin >> x[i] >> y[i];
    }

    int area = 0;

    for (int i = 0; i < n; i++) {
        area += x[i] * y[(i + 1) % n];
        area -= y[i] * x[(i + 1) % n];
    }

    cout << abs(area) << endl;

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/geometry/Polygon_Area.py

def calculate_area(n, vertices):
    area = 0
    for i in range(n):
        x1, y1 = vertices[i]
        x2, y2 = vertices[(i + 1) % n]
        area += x1 * y2 - x2 * y1
    return abs(area) // 2

n = int(input())
vertices = []
for _ in range(n):
    x, y = map(int, input().split())
    vertices.append((x, y))

area = calculate_area(n, vertices)
print(2 * area)
